dockstore-yaml-proposals/README.md

```diff
@@ -1,12 +1,195 @@
+* Xena Hub
+* JBrowse
+* (not ready) [Bravo](https://github.com/statgen/bravo) -- Variant browser
+* (not ready) generick8s -- this is a Kubernetes guest book example application. Using it until I can find a bioinformatics, K8s service.
 
-Currently we are just actively iterating over Xena.
+
+### Terms
+
+* Service -- A long running process, such as a web app or interactive application.
+* .dockstore.yml -- A configuration file that describes a service.
+* Launcher or Service Launcher -- An application that launches the service, using the information provided in the .dockstore.yml and the input parameter file. Examples: Leonardo, Dockstore CLI.
+* User -- the consumer of the service.
+* Input parameter file -- allows the user and/or launcher to specify inputs to the service, e.g., what genomics data to download, what environment variables to set.
+
+### The .dockstore.yml
+
+The .dockstore.yml is similar to .travis.yml and .circle-ci/config.yml; it is a file committed to a source code repository, and it describes the service. The
+launcher uses the file to stand up the service. The .dockstore.yml in also used by Docsktore to surface the service in Dockstore.
+
+The schema of the .dockstore.yml follows. For now, we will describe by example; a formal schema may follow.
+
+```
+# The dockstore.yml version; 1.1 is required for services
+version: 1.1
+
+# A required key named service
+service:
+  # The type is required, and can be docker-compose, kubernetes, or helm.
+  type: docker-compose
+
+  # The name of the service, required
+  name: UCSC Xena Browser
+
+  # The author, optional
+  author: UCSC Genomics Institute
+
+  # An optional description
+  description: |
+    The UCSC Xena browser is an exploration tool for public and private,
+    multi-omic and clinical/phenotype data.
+    It is recommended that you configure a reverse proxy to handle HTTPS
+
+  # These are files the Dockstore will index. They will be directly downloadable from Dockstore. Wildcards are not supported.
+  files:
+     - docker-compose.yml
+     - README.md
+     - stand_up.sh
+     - load.sh
+     - port.sh
+     - stop.sh
+     - healthcheck.sh
+
+  scripts:
+    # The scripts section has 8 pre-defined keys. 5 are run in a defined order,
+    # the other 3 are utilities that can be invoked by the service launcher.
+    # Only 3 of the keys must exist and have values: start, port, and stop.
+    # The keys' values, if present, must be script files that are indexed (in the files section, above).
+
+    # The service launcher will execute the scripts in the following order. All steps other than start are optional,
+    # and if they are missing or have no value specified, will be ignored.
+    #
+    # 1. preprovision -- Invoked before anything has been done.
+    # 2. prestart -- Executed after data has been downloaded locally (see the data section)
+    # 3. start -- Starts up the service.
+    # 4. poststart -- After the service has been started
```

Comment:
Maybe some examples of what this should be used for, same with 1

Reply:
> Maybe some examples of what this should be used for, same with 1

I'm having a hard  time thinking of concrete cases. :(  But it seems like they should be there, just in case. Will think some more, and add something.

Reply:
Ended up removing poststart; I think there could be a use case for preprovision, some onetime initialization before launching the service, so I've left that.